Gems/SceneProcessing/Code/Source/Generation/Components/MeshOptimizer/MeshOptimizerComponent.cpp

```diff
@@ -261,6 +265,48 @@ namespace AZ::SceneGenerationComponents
         ).empty();
     }
 
+    static ICustomPropertyData::PropertyMap& FindOrCreateCustomPropertyData(
+        Containers::SceneGraph& graph, const Containers::SceneGraph::NodeIndex& nodeIndex)
+    {
+        NodeIndex customPropertyIndex =
+            SceneAPI::Utilities::GetImmediateChildOfType(graph, nodeIndex, azrtti_typeid<ICustomPropertyData>());
+
+        if (!customPropertyIndex.IsValid())
+        {
+            // If no custom property data node exists, insert one
+            AZStd::shared_ptr<SceneData::GraphData::CustomPropertyData> createdCustumPropertyData =
+                AZStd::make_shared<SceneData::GraphData::CustomPropertyData>();
+            customPropertyIndex = graph.AddChild(nodeIndex, "custom_properties", AZStd::move(createdCustumPropertyData));
+        }
+
+        ICustomPropertyData* customPropertyDataNode =
+            azrtti_cast<ICustomPropertyData*>(graph.GetNodeContent(customPropertyIndex).get());
+
+        return customPropertyDataNode->GetPropertyMap();
+    }
+
+    static bool HasOptimizedMeshNode(Containers::SceneGraph& graph, ICustomPropertyData::PropertyMap& propertyMap)
+    {
+        // Now look up the optimized index
+        if (auto iter = propertyMap.find(SceneAPI::Utilities::OptimizedMeshPropertyMapKey); iter != propertyMap.end())
+        {
+            const auto& [key, anyNodeIndex] = *iter;
+            if (!anyNodeIndex.empty() && anyNodeIndex.is<NodeIndex>())
+            {
+                NodeIndex optimizedMeshIndex = AZStd::any_cast<NodeIndex>(anyNodeIndex);
+                AZ_TracePrintf(
```

Comment:
I find this trace info is not connected to the function. It's probably better to have this information at where the function is called.